opentracing/src/main/java/opentracing/Tracer.java

```diff
@@ -13,25 +13,41 @@
  */
 package opentracing;
 
+
 /**
- * Tracer is a simple, thin interface for Span creation.
+ * Tracer is a simple, thin interface for Span creation, and Span propagation into different transport formats.
  */
 public interface Tracer {
 
   /**
-   * Create, start, and return a new Span with the given `operationName`, all without specifying a
-   * parent Span that can be used to incorporate the newly-returned Span into an existing trace.
+   * Create, start, and return a new Span with the given `operationName`.
+   * An optional parent Span can be specified used to incorporate the newly-returned Span into an existing trace.
    *
    * <p>Example:
    * <pre>{@code
    * Tracer tracer = ...
-   * Span feed = tracer.startTrace("GetFeed");
-   * Span http = tracer.startTrace("HandleHTTPRequest")
+   * Span feed = tracer.startTrace("GetFeed", null);
+   * Span http = tracer.startTrace("HandleHTTPRequest", feed)
    *                   .setTag("user_agent", req.UserAgent)
    *                   .setTag("lucky_number", 42);
    * }</pre>
    */
-  Span startTrace(String operationName);
+  Span startSpan(String operationName, /* @Nullable */ Span parent);
+
+  /** Takes two arguments:
+   *    a Span instance, and
+   *    a “carrier” object in which to inject that Span for cross-process propagation.
+   *
+   */
+  <T> void inject(Span span, T carrier);
```

Comment:
I've advocated that the format parameter be optional, not removed.
I just don't feel for adding public methods that we're forever bound to prematurely.

> The intermediate methods (and types) are not relevant here …

What I meant by that is… the problem introduced here regarding formats, was solved with the tracer.getInjector(..) and tracer.getExtractor(..)

(the tying of generics around carrier did over emphased the binding, but I would have removed that, or permitted tracer.getInjector(string) calls so to satisfy the case of overlapping carrier types. once again i would have advocated for it when needed.)

Taking a look at the current [java Public API](https://github.com/opentracing/opentracing-java/tree/mck/delegation-design-0/opentracing/src/main/java/opentracing) I'm quite content with how it has simplified and become incredibly easy to quickly comprehend. It will grow of course, but this is a great place for "1.0" in my opinion. Post a released 1.0 hopefully, with an api that everyone can quickly grab uncluttered by specific use-cases, we'll see further implementing into the different tracing instrumentations out there, which can provide us with even more use-cases and corner-cases.